spec/lib/nerve/reporter_zookeeper_spec.rb

```diff
@@ -29,5 +31,103 @@
     reporter.report_up
     reporter.stop
   end
+
+  context "when reporter has been up and there is connection issue" do
+    before(:each) do
+      allow(zk).to receive(:close!)
+      allow(zk).to receive(:connected?).and_return(true)
+      expect(zk).to receive(:exists?) { "zk_path" }.and_return(false)
+      expect(zk).to receive(:mkdir_p) { "zk_path" }
+      expect(zk).to receive(:create) { "full_path" }
+      allow(ZK).to receive(:new).and_return(zk)
+
+      @reporter = Nerve::Reporter::Zookeeper.new(subject)
+      @reporter.start
+      @reporter.report_up
+    end
+
+    after(:each) do
+      # reset the class variable to avoid mock object zk leak
+      Nerve::Reporter::Zookeeper.class_variable_set(:@@zk_pool, {})
+      Nerve::Reporter::Zookeeper.class_variable_set(:@@zk_pool_count, {})
+    end
+
+    it 'returns false on ping? when zk is disconnected' do
```

Comment:
you're not using `context` in the most clear way. ideally, you should have multiple nested contexts, and tests inside that show how the behavior is different within the `context`. for example:

```ruby
context "when the reporter is up" do
  before do
    allow(zk).to receive(:connected?).and_return(true)
  end

  it 'pings successfully' do
    expect(@reporter.ping?).to be_falsey
  end

  context 'and there is a connection issue' do
    before { expect(zk).to receive(:connected?).and_return(false)

    it 'fails to ping?' do
      expect(@reporter.ping?).to be_falsey
    end
  end
end
```

by separating your test setup into the `context` block, you make it clear what's different about each `context` (situation) and what's different about the behavior of nerve in each `context`